File: test-accesstoken-pkg/ts/test-docs-setup.ts
/**
 * Mirrors website/public/getting-started/setup.md against @imbrace/sdk@1.0.4
 * (npm) — Access Token auth.
 */
import "dotenv/config"
import { ImbraceClient } from "@imbrace/sdk"

const accessToken    = process.env.IMBRACE_ACCESS_TOKEN
const organizationId = process.env.IMBRACE_ORGANIZATION_ID
const baseUrl        = process.env.IMBRACE_GATEWAY_URL ?? "https://app-gatewayv2.imbrace.co"

if (!accessToken || !organizationId) { console.error("Missing creds"); process.exit(1) }

let pass = 0, fail = 0, skip = 0
const fails: string[] = []

async function step(label: string, fn: () => Promise<unknown>) {
  process.stdout.write(`  • ${label} ... `)
  try { const r = await fn(); console.log(`✓ ${JSON.stringify(r ?? {}).slice(0, 90)}`); pass++ }
  catch (err: any) { console.log(`✗ [${err?.statusCode ?? err?.message}]`); fail++; fails.push(`${label} → ${err?.statusCode ?? err?.message}`) }
}
function skipped(label: string, reason: string) { console.log(`  - ${label}  ⏭ ${reason}`); skip++ }

console.log(`\n━━━ DOCS: getting-started/setup.md — auth: ACCESS TOKEN (npm @imbrace/sdk@1.0.4) ━━━\n`)

console.log("══ §Init by env preset ══")
await step("new ImbraceClient({ accessToken, env: 'stable' }).platform.getMe()",
  () => new ImbraceClient({ accessToken, organizationId, env: "stable" as any, timeout: 8_000 }).platform.getMe())

console.log("══ §Init by baseUrl ══")
await step("new ImbraceClient({ accessToken, baseUrl }).platform.getMe()",
  () => new ImbraceClient({ accessToken, organizationId, baseUrl, timeout: 8_000 }).platform.getMe())

console.log("══ §services override (construct only) ══")
await step("new ImbraceClient({ services: {...} }) constructs ok",
  async () => {
    const c = new ImbraceClient({
      accessToken, organizationId, env: "develop" as any, timeout: 1_000,
      services: { aiAgent: "http://localhost:4000/ai-agent" } as any,
    })
    return { ok: typeof c === "object" }
  })

console.log("══ §Hello-world calls ══")
await step("client.channel.list()",
  () => new ImbraceClient({ accessToken, organizationId, baseUrl, timeout: 8_000 }).channel.list({} as any))
skipped("messages.send", "destructive")
skipped("ai.complete / ai.stream / aiAgent.streamChat", "covered in test-docs-ai-agent.ts")

console.log("══ §OTP login flow ══")
skipped("requestOtp / loginWithOtp", "destructive")

console.log(`\n━━━ Summary (setup / Access Token) ━━━`)
console.log(`pass=${pass}  fail=${fail}  skip=${skip}`)
if (fails.length) { console.log("Failures:"); fails.forEach(f => console.log(`  - ${f}`)) }
process.exit(fail > 0 ? 1 : 0)

File: test-accesstoken-pkg/py/test_docs_setup.py
"""Mirrors website/public/getting-started/setup.md against `imbrace==1.0.4`
(PyPI) — Access Token auth.
"""
from __future__ import annotations
import os, sys, json, asyncio
from dotenv import load_dotenv
load_dotenv()
from imbrace import ImbraceClient, AsyncImbraceClient

ACCESS_TOKEN = os.environ.get("IMBRACE_ACCESS_TOKEN"); ORG_ID = os.environ.get("IMBRACE_ORGANIZATION_ID")
GATEWAY = os.environ.get("IMBRACE_GATEWAY_URL", "https://app-gatewayv2.imbrace.co")
if not ACCESS_TOKEN or not ORG_ID:
    print("Missing creds"); sys.exit(1)

passed = failed = skipped_n = 0; failures: list[str] = []


def step(label, fn):
    global passed, failed
    sys.stdout.write(f"  - {label} ... "); sys.stdout.flush()
    try:
        r = fn(); print(f"OK {json.dumps(r, default=str)[:90]}"); passed += 1
    except Exception as e:
        print(f"FAIL [{str(e)[:80]}]"); failed += 1; failures.append(f"{label} -> {str(e)[:80]}")


def skip(label, reason):
    global skipped_n
    print(f"  - {label}  SKIP: {reason}"); skipped_n += 1


print(f"\n=== DOCS: getting-started/setup.md - auth: ACCESS TOKEN (PyPI imbrace==1.0.4) ===\n")

print("== §Init by env preset (env='stable') ==")
step("ImbraceClient(access_token=..., env='stable').platform.get_me()",
     lambda: ImbraceClient(access_token=ACCESS_TOKEN, organization_id=ORG_ID, env="stable", timeout=8).platform.get_me())

print("== §Init by gateway URL ==")
step("ImbraceClient(access_token=..., gateway=...).platform.get_me()",
     lambda: ImbraceClient(access_token=ACCESS_TOKEN, organization_id=ORG_ID, gateway=GATEWAY, timeout=8).platform.get_me())

print("== §services override (construct only) ==")


def _services_override():
    c = ImbraceClient(
        access_token=ACCESS_TOKEN, organization_id=ORG_ID, env="develop", timeout=1,
        services={"ai_agent": "http://localhost:4000/ai-agent"},
    )
    return {"ok": c is not None}


step("ImbraceClient(services={...}) constructs ok", _services_override)

print("== §AsyncImbraceClient hello world ==")


def _async_hello():
    async def _run():
        async with AsyncImbraceClient(access_token=ACCESS_TOKEN, organization_id=ORG_ID, gateway=GATEWAY, timeout=8) as ac:
            try:
                return await ac.platform.get_me()
            except Exception as e:
                return {"err": str(e)[:60]}
    return asyncio.run(_run())


step("async with AsyncImbraceClient(access_token=...) as ac: ac.platform.get_me()", _async_hello)

print("== §Hello-world calls ==")
client = ImbraceClient(access_token=ACCESS_TOKEN, organization_id=ORG_ID, gateway=GATEWAY, timeout=8)
step("client.channel.list()", lambda: client.channel.list())
skip("messages.send / ai.complete / ai_agent.stream_chat", "destructive or covered elsewhere")

print("== §OTP login flow ==")
skip("request_otp / login_with_otp", "destructive")

print(f"\n=== Summary (setup / Access Token) ===")
print(f"pass={passed}  fail={failed}  skip={skipped_n}")
if failures:
    print("Failures:")
    for f in failures: print(f"  - {f}")
sys.exit(1 if failed > 0 else 0)
